raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -370,8 +363,52 @@ private void maybeFireLeaderChange() {
         }
     }
 
-    @Override
-    public void initialize() {
+    public void initialize(
+        Map<Integer, InetSocketAddress> voterAddresses,
+        String listenerName,
+        QuorumStateStore quorumStateStore,
+        Metrics metrics
+    ) {
+        partitionState = new KRaftControlRecordStateMachine(
+            Optional.of(VoterSet.fromAddressSpecs(listenerName, voterAddresses)),
```

Comment:
nit: maybe we should rename the method `fromInetSocketAddresses` or something like that?

Reply:
Fixed.